### PR TITLE
[CNDIT-1511] Initial Code Commit

### PR DESCRIPTION
**Pull request by rshanmugam-enquizit**
### Notes
This PR includes the initial commit for this repo for all the NRT micro services.

### JIRA
- https://cdc-nbs.atlassian.net/browse/CNDIT-1511

### Testing
- Ran the command `./gradlew clean build jacocoTestReport` from the root directory and was able to see the test result files being generated.
- Github actions are passing except the sonar scan which will be done by the foundation team.
- Ran the unit tests individually and all of them passed.

### CODE REVIEW

File: .github/workflows/Build-and-deploy-reporting-services.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
+name: Build and push data-reporting-service image to ECR
+on:
+  push:
+    branches:
+      - main
+      - master
+      - rel-**
+#       Uncomment the following line only to test the build deploy from private branches.
+      - CNDIT-*
```
Comment (alog-enquizit) [inline .github/workflows/Build-and-deploy-reporting-services.yaml:9]: Make sure to comment this out when you merge to main

File: .github/workflows/Build-and-deploy-reporting-services.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      environment_classifier: SNAPSHOT
+      java_version: "17"
+    secrets:
+      CDC_NBS_SANDBOX_SHARED_SERVICES_ACCOUNTID: ${{secrets.CDC_NBS_SANDBOX_SHARED_SERVICES_ACCOUNTID}}
+      ECR_REPO_BASE_NAME: ${{secrets.ECR_REPO_BASE_NAME}}
+      GIT_USER_EMAIL: ${{secrets.GIT_USER_EMAIL}}
+      GIT_USER_NAME: ${{secrets.GIT_USER_NAME}}
+      HELM_TOKEN: ${{secrets.HELM_TOKEN}}
+
+  call-update-helm-for-person-reporting-microservice:
```
Comment (alog-enquizit) [inline .github/workflows/Build-and-deploy-reporting-services.yaml:37]: Nice 👍

**Comment by alog-enquizit** [comment]
LDFData Reporting Service image build is missing.
Observation Reporting Service image build is missing.

**Review by alog-enquizit — APPROVED** [review]
LGMT 👍 Thanks for migrating to the new repo.

File: .github/workflows/Build-and-deploy-reporting-services.yaml (hunk 2 above)
Comment (rshanmugam-enquizit) [reply .github/workflows/Build-and-deploy-reporting-services.yaml:37]: :)

File: .github/workflows/Build-and-deploy-reporting-services.yaml (hunk 1 above)
Comment (rshanmugam-enquizit) [reply .github/workflows/Build-and-deploy-reporting-services.yaml]: Done

**Pull request merged**